src/gleam_stdlib.mjs

```diff
@@ -220,6 +220,30 @@ export function length(data) {
   return data.length;
 }
 
+export function string_slice(string, idx, len) {
```

Comment:
mmh this is interesting! Avoiding all the extra work of splitting the entire string, creating a list of graphemes and walking through it when slicing seems compelling to me

Reply:
This is pretty much what the Erlang version does 👍

Reply:
Added missing early out when `len <= 0 || idx >= string.length`.